apm-agent-core/src/test/java/co/elastic/apm/agent/objectpool/TestObjectPoolFactory.java

```diff
@@ -79,7 +79,7 @@ public void checkAllPooledObjectsHaveBeenRecycled() {
                     // silently ignored
                 }
             }
-        } while (retry-- > 0 && hasSomethingLeft);
+        } while (--retry > 0 && hasSomethingLeft);
```

Comment:
The postfix decrement caused `retry` to become `-1` at the end of the loop, effectively causing the `if` condition below to always evaluate to false.